### ISSUE TITLE
Usage of softmax prior to loss functions?

### ISSUE THREAD

**Issue by shreyaskamathkm** [haitaobiyao/MCNet#2]
Hi,

I was going through your code and found that you have utilized [Crossentropy loss](https://github.com/haitaobiyao/MCNet/blob/ed79ad44c9320697a13a1ee2b29ce79037741b2a/TorchSeg-MCNet/model/MCNet/scut.mcnet.R101/train.py#L49) as a loss function for training the network.  In the network function, the input to the loss function has already passed through a [F.log_softmax](https://github.com/haitaobiyao/MCNet/blob/ed79ad44c9320697a13a1ee2b29ce79037741b2a/TorchSeg-MCNet/model/MCNet/scut.mcnet.R101/network.py#L54) function.  

The nn.Crossentropy loss function provided by [pytorch](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html) combines the LogSoftmax and NLLLoss. Could you please confirm if this was intentional?

Thank you,
SK

**Comment by haitaobiyao** [comment]
> Hi,
> 
> I was going through your code and found that you have utilized [Crossentropy loss](https://github.com/haitaobiyao/MCNet/blob/ed79ad44c9320697a13a1ee2b29ce79037741b2a/TorchSeg-MCNet/model/MCNet/scut.mcnet.R101/train.py#L49) as a loss function for training the network. In the network function, the input to the loss function has already passed through a [F.log_softmax](https://github.com/haitaobiyao/MCNet/blob/ed79ad44c9320697a13a1ee2b29ce79037741b2a/TorchSeg-MCNet/model/MCNet/scut.mcnet.R101/network.py#L54) function.
> 
> The nn.Crossentropy loss function provided by [pytorch](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html) combines the LogSoftmax and NLLLoss. Could you please confirm if this was intentional?
> 
> Thank you,
> SK
Thank you for pointing this out. In fact, whether to use softmax prior has almost no effect on the experimental results. Therefore, we deleted softmax prior and re-uploaded network.py.

**Comment by haitaobiyao** [comment]
> Hi,
> 
> I was going through your code and found that you have utilized [Crossentropy loss](https://github.com/haitaobiyao/MCNet/blob/ed79ad44c9320697a13a1ee2b29ce79037741b2a/TorchSeg-MCNet/model/MCNet/scut.mcnet.R101/train.py#L49) as a loss function for training the network. In the network function, the input to the loss function has already passed through a [F.log_softmax](https://github.com/haitaobiyao/MCNet/blob/ed79ad44c9320697a13a1ee2b29ce79037741b2a/TorchSeg-MCNet/model/MCNet/scut.mcnet.R101/network.py#L54) function.
> 
> The nn.Crossentropy loss function provided by [pytorch](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html) combines the LogSoftmax and NLLLoss. Could you please confirm if this was intentional?
> 
> Thank you,
> SK

**Comment by haitaobiyao** [comment]
> Hi,
> 
> I was going through your code and found that you have utilized [Crossentropy loss](https://github.com/haitaobiyao/MCNet/blob/ed79ad44c9320697a13a1ee2b29ce79037741b2a/TorchSeg-MCNet/model/MCNet/scut.mcnet.R101/train.py#L49) as a loss function for training the network. In the network function, the input to the loss function has already passed through a [F.log_softmax](https://github.com/haitaobiyao/MCNet/blob/ed79ad44c9320697a13a1ee2b29ce79037741b2a/TorchSeg-MCNet/model/MCNet/scut.mcnet.R101/network.py#L54) function.
> 
> The nn.Crossentropy loss function provided by [pytorch](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html) combines the LogSoftmax and NLLLoss. Could you please confirm if this was intentional?
> 
> Thank you,
> SK
